### PR TITLE
Saved Reports now uses standarized date picker

### PR DESCRIPTION
**Pull request by mjriley**
## Technical Summary
[Associated Ticket](https://dimagi-dev.atlassian.net/browse/SAAS-12917). This is an initial fix for the specific problem. I'd like to address other instances where the old date picker is used, but wanted to make sure that my approach here was correct first.

### Safety story
Performed local testing. Some of the validation fixes I included are probably worthy of tests, but I'm not sure of our javascript testing environment.

### Automated test coverage

See above.

### QA Plan

No QA necessary.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/reports/static/reports/js/report_config_models.js [hunk 1]
```diff
@@ -4,20 +4,22 @@ hqDefine("reports/js/report_config_models", [
...
-    'jquery-ui/ui/widgets/datepicker',
+    'bootstrap-daterangepicker/daterangepicker',
 ], function (
     $,
     ko,
     _,
     googleAnalytics,
-    standardHQReportModule
+    standardHQReportModule,
+    dateRangePicker  // jshint ignore:line
```
[the remark this answers is not in the document]
Comment (mjriley) [reply corehq/apps/reports/static/reports/js/report_config_models.js:14]: any idea what the proper way to dodge this warning is? I think not listing the module in the include list is error-prone.

[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/reports/static/reports/js/report_config_models.js:14]: Yeah, the module needs to be there. You just need different syntax to ignore it, check out https://eslint.org/docs/2.13.1/user-guide/configuring#disabling-rules-with-inline-comments

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/reports/static/reports/js/report_config_models.js [hunk 2]
```diff
@@ -29,22 +31,34 @@ hqDefine("reports/js/report_config_models", [
         });
 
         self.validate = function () {
-            var date_range = self.date_range(),
+            let dateRange = self.date_range(),
                 error = false;
 
+            let errorMessage = gettext("Some required fields are missing. Please complete them before saving.");
```
[the remark this answers is not in the document]
Comment (mjriley) [reply corehq/apps/reports/static/reports/js/report_config_models.js:37]: Yeah, I think this is out of scope for this specific ticket. This blanket error message was applied to all errors before, when it wasn't throwing any error if the end date occurred before the start date. When I added handling for that, it felt inappropriate to just say that a field was missing. I don't disagree with adding more specific error messages for each field, but I'd like to keep this PR focused on the date picker.

File: corehq/apps/reports/templates/reports/partials/save_reports_modal.html [hunk 3]
```diff
@@ -10,8 +10,7 @@ <h4 class="modal-title" data-bind="text: modalTitle"></h4>
       </div>
       <form class="form form-horizontal">
         <div class="modal-body">
-          <div class="alert alert-danger" data-bind="visible: error">
-            {% trans "Some required fields are missing. Please complete them before saving." %}
+          <div class="alert alert-danger" id="error_message" data-bind="visible: error, text: errorMessage">
```
[the remark this answers is not in the document]
Comment (mjriley) [reply corehq/apps/reports/templates/reports/partials/save_reports_modal.html:13]: Thanks. This was a relic before I figured out the text binding

**Review by orangejenny — APPROVED** [review]

**Review by biyeun — APPROVED** [review]

**Pull request merged**